src/Workleap.DomainEventPropagation.Subscription.PullDelivery/EventPuller.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Azure.Messaging.EventGrid.Namespaces;
+using Microsoft.Extensions.Azure;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Options;
+
+namespace Workleap.DomainEventPropagation;
+
+internal class EventPuller : BackgroundService
+{
+    private readonly EventGridReceptionChannel[] _eventGridReceptionChannels;
+
+    public EventPuller(
+        IEnumerable<EventGridClientDescriptor> clientDescriptors,
+        IAzureClientFactory<EventGridClient> eventGridClientFactory,
+        IOptionsMonitor<EventPropagationSubscriptionOptions> optionsMonitor)
+    {
+        this._eventGridReceptionChannels = clientDescriptors.Select(descriptor =>
+            new EventGridReceptionChannel(
+                optionsMonitor.Get(descriptor.Name).TopicName,
+                optionsMonitor.Get(descriptor.Name).SubscriptionName,
+                eventGridClientFactory.CreateClient(descriptor.Name))).ToArray();
+    }
+
+    protected override Task ExecuteAsync(CancellationToken stoppingToken)
+    {
+        return Task.WhenAll(this._eventGridReceptionChannels.Select(channel => Task.Run(() => StartReceivingEventsAsync(channel, stoppingToken), stoppingToken)));
```

Comment:
You _may_ not need this:
```suggestion
        return Task.WhenAll(this._eventGridReceptionChannels.Select(channel => StartReceivingEventsAsync(channel, stoppingToken), stoppingToken));
```

Reply:
Nope, if we remove this we end up with only one task that run at the time (behavior is unit tested). 
If we want to read from every topics in parallel, we need to force its execution on the thread pool